### PR TITLE
WIP: *: Add "AWS permission" annotations

### PR DESCRIPTION
**Pull request by wking**
So we can answer "what permissions does the installer need" with:

```console
$ git grep 'AWS permission:' | sed 's/.*AWS permission: //' | sort | uniq
ec2:DeleteDhcpOptions
ec2:DescribeInstances
...
```

This will also let us distinguish between create and destroy permissions, so we could pre-flight destroy calls (we probably want to continue to check for both before installing).

WIP because I still have some annotations to drop into the Terraform, and because I'm guessing in a few places and would like to have CI results to back me up (probably by creating more tailored perms for install/destroy in the CI templates).

### CODE REVIEW

**Comment by wking** [comment]
CC @abhinavdahiya, @kalexand-rh.  Things like [`ec2:UpdateSecurityGroupRuleDescriptionsIngress`][1] are new, so we probably want some wiggle-words in permission docs until we have CI to back us up.

[1]: https://github.com/openshift/installer/pull/1752/files#diff-c421fc80402da42875570d21c071ada2R12

File: hack/generate-aws-permissions.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
```
Comment (abhinavdahiya) [inline hack/generate-aws-permissions.sh]: needs to use `go generate`

Comment (wking) [reply hack/generate-aws-permissions.sh]: > needs to use `go generate`

Is calling out to the shell script acceptable, just with a `go generate` trigger?  Or do I need to port the regexp extractor from shell to Go too?

Comment (abhinavdahiya) [reply hack/generate-aws-permissions.sh]: former

Comment (wking) [reply hack/generate-aws-permissions.sh]: Added a `go:generate` comment with 78f39f075 -> eb892c7d7.

**Comment by abhinavdahiya** [comment]
closing due to inactivity. Please reopen if needed.

/close

**Pull request closed without merging**